packages/office-ui-fabric-react/src/components/Sticky/Sticky.tsx

```diff
@@ -113,12 +118,18 @@ export class Sticky extends BaseComponent<IStickyProps, IStickyState> {
     }
 
     const { isStickyTop, isStickyBottom } = this.state;
-    return (
-      isStickyTop !== nextState.isStickyTop ||
-      isStickyBottom !== nextState.isStickyBottom ||
-      this.props.stickyPosition !== nextProps.stickyPosition ||
-      this.props.children !== nextProps.children
-    );
+
+    if (this._nonStickyContent && this._nonStickyContent.current && this._placeHolder && this._placeHolder.current) {
+      return (
+        isStickyTop !== nextState.isStickyTop ||
+        isStickyBottom !== nextState.isStickyBottom ||
+        this.props.stickyPosition !== nextProps.stickyPosition ||
+        this.props.children !== nextProps.children ||
+        this._nonStickyContent.current.offsetHeight !== this._placeHolder.current.offsetHeight
+      );
+    }
```

Comment:
This condition looks very restrict, most cases it will just return false, is this expected?

Reply:
Ok, if the reason to add this if statement is to make sure this new condition will not complain about undefined issue, then the proper fix should be like below. That will make sure later on if anything change, we will still go through the following code but not just return false. 

    return (
      isStickyTop !== nextState.isStickyTop ||
      isStickyBottom !== nextState.isStickyBottom ||
      this.props.stickyPosition !== nextProps.stickyPosition ||
      this.props.children !== nextProps.children ||
      (this._nonStickyContent && this._nonStickyContent.current && this._placeHolder && this._placeHolder.current &&
         this._nonStickyContent.current.offsetHeight !== this._placeHolder.current.offsetHeight
      )
    );


---
In reply to: [232749338](https://github.com/OfficeDev/office-ui-fabric-react/pull/6937#discussion_r232749338) [](ancestors = 232749338)